src/automerge.js

```diff
@@ -95,9 +96,10 @@ function getAllChanges(doc) {
   return getChanges(init(), doc)
 }
 
-function applyChanges(doc, changes) {
+function applyChanges(doc, changes, options = {}) {
```

Comment:
why is there an options map needed. Could we just pass the callback directly?

Reply:
I prefer the interface @ept suggests here - the options map makes it easier to add the change API further in the future. It also makes the code calling this function more readable.

Reply:
I agree that readability would improve on the call side. But I think that's caused by missing labeled arguments in JavaScript which we should not try to fix.

When just reading `function applyChanges(doc, changes, options = {})` I would expect to pass a document, some changes and a map which only contains key **value** pair, which help configuring the change method. I would not expect to pass references into `options` which provide me with the resulting patches.

On the other hand when reading `function applyChanges(doc, changes, patchCallback)` it is clear to me that I can pass a document, changes and a callback which provides some kind of patches. (It is also future prove to just add another parameter)

This is not really a strong opinion, so I am happy to go with options map as well :)

Reply:
I'm with @HerbCaudill here. I think this kind of options hash is common in JavaScript, since it gives us something a bit like named parameters in the absence of language support, and it allows us to add further options in the future without breaking existing code. Moreover, lots of existing functions in the Automerge API (`init`, `from`, `load`, `change`, etc.) already take an options hash. I don't see a problem with passing a function as part of an options hash: in JS, functions are values like any other.